src/sas/sascalc/fit/expression.py

```diff
@@ -44,40 +44,79 @@ def model_update():
 """
 from __future__ import print_function
 
+from copy import copy
 import math
 import re
+from keyword import iskeyword
+
+def standard_symbols(context={}):
+    symbols = {}
+    symbols.update(math.__dict__)
+    symbols.update(dict(arcsin=math.asin, arccos=math.acos,
+                        arctan=math.atan, arctan2=math.atan2))
+    symbols.update(context)
+    symbols['id'] = id
+    return symbols
+
+def _check_syntax(target, expr, html = False):
+    try:
+        compile(expr, expr, "exec")
+    except SyntaxError as exc:
+        if html:
+            constraint = "% s = % s" % (target, expr)
+            text = exc.text.strip("\n")
+            highlighted_text = text[:exc.offset-1] + "<b>" + \
+                               text[exc.offset-1:] + "</b>"
+            constraint = constraint.replace(text, highlighted_text)
```

Comment:
This is confusing since it is not obvious that `text` is a substring of `expr`.  Some comments are in order.

If the error is a missing close parenthesis, then it will show up after the end of text, and no error will be indicated.

Multiline syntax will not be formatted correctly without `<pre>`.

The `% s` syntax in the format string is valid but unusual. Should be just `%s`.

So maybe:
```python
if "\n" in expr:
    # Multiline expression. Be lazy and just show line, col for syntax error since this should never happen.
    return [f"Syntax error on line {exc.lineno} column {exc.offset} for {target}:\n<pre>\n{expr}\n</pre>"]

if exc.offset > len(expr):
     # Single line expression with error after the expression. Probably missing a closing paren, but it could
     # also be that the expression ends with an operator such as "3+4+"
    return [f"Expression `{target} = {expr}` is not complete."]

# Single line expression. Wrap the tail of the expr after the syntax error in <b>...</b>.
# exc.lineno=1, exc.text = expr+"\n", and exc.offset = location of the syntax error in expr.
return [f"Syntax error in expression '{target} = {expr[:exc.offset-1]}<b>{expr[exc.offset-1:]}</b>'"]
```